lib/components/segmented-control/segment.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { forwardRef } from "react";
+import clsx from "clsx";
+import {
+    LabelPairedPlaceholderLgRegularIcon,
+    LabelPairedPlaceholderMdRegularIcon,
+    LabelPairedPlaceholderSmRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../Typography";
+import { SegmentedControlProps } from "./base";
+
+interface SegmentProps {
+    className?: string;
+    icon?: string | React.ReactNode;
+    isDisabled?: boolean;
+    isSelected?: boolean;
+    label?: string;
+    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+    onKeyDown?: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+    size?: SegmentedControlProps["size"];
+}
+
+const placeholder = {
+    sm: <LabelPairedPlaceholderSmRegularIcon />,
+    md: <LabelPairedPlaceholderMdRegularIcon />,
+    lg: <LabelPairedPlaceholderLgRegularIcon />,
+};
+
+export const Segment = forwardRef<HTMLButtonElement, SegmentProps>(
+    (
+        {
+            className,
+            icon,
+            isDisabled,
+            isSelected,
+            label,
+            onClick,
+            onKeyDown,
+            size,
+        },
+        ref,
+    ) => {
+        const Icon =
+            icon === "placeholder"
+                ? placeholder[size as keyof typeof placeholder]
+                : icon;
+        return (
+            <button
+                className={clsx(
+                    "item",
+                    isDisabled && "disabled",
+                    isSelected && "selected",
+                    className,
+                )}
+                onClick={isDisabled ? undefined : onClick}
+                onKeyDown={onKeyDown}
+                ref={ref}
```

Comment:
Why are you avoiding using `disabled` button attribute? 
It's very natural + will help to get rid of checks inside of `onClick` and `onKeyDown`. 
Moreover, you also can use `:disable` in css instead of classname.

Reply:
It will be also better for screen-readers, as they will be notified, that button was disabled

Reply:
@kate-deriv i'm use both CSS and JS for this. We should not get rid of checks inside of onClick and onKeyDown because CSS is something that any user can alter in his Browser Dev Tools. CSS is for beauty (pointer-events) while JS should always be handling the logic (which events are allowed/disabled).

Reply:
good point :) that's good i changed `div` to `button` and able to use it disabled attribute now👍🏻